lang/LangPrimSource/SC_AlsaMIDI.cpp

```diff
@@ -1008,6 +1012,18 @@ int prSendSysex(VMGlobals *g, int numArgsPushed)
 	return sendMIDISysex(outputIndex, uid, packet->size, packet->b);
 }
 
+int prGetMIDIClientID(VMGlobals *g, int numArgsPushed)
+{
+  
+  PyrSlot* args = g->sp;
+  if (!gMIDIClient.mHandle) return errFailed;
+
+  
+  SetInt(args, gMIDIClient.mClientID );
+  
+  return errNone;
+}
+
```

Comment:
This only handles the linux case, no? There needs to be an OSX and Win implementation of this primitive. The OSX would look something like this (I think...following SC_CoreMIDI.cpp):

		SInt32 id;
		MIDIObjectGetIntegerProperty(gMIDIClient, kMIDIPropertyUniqueID, &id);